sdk/pinocchio/src/entrypoint/lazy.rs

```diff
@@ -241,10 +233,35 @@ impl InstructionContext {
     /// before reading all accounts will result in undefined behavior.
     #[inline(always)]
     pub unsafe fn program_id_unchecked(&self) -> &Pubkey {
-        let data_len = *(self.input.add(self.offset) as *const usize);
-        &*(self
-            .input
-            .add(self.offset + core::mem::size_of::<u64>() + data_len) as *const Pubkey)
+        let data_len = *(self.input as *const usize);
+        &*(self.input.add(core::mem::size_of::<u64>() + data_len) as *const Pubkey)
+    }
+
+    /// Read an account from the input buffer.
+    ///
+    /// This can only be called with a buffer that was serialized by the runtime as
+    /// it assumes a specific memory layout.
+    #[allow(clippy::cast_ptr_alignment, clippy::missing_safety_doc)]
+    #[inline(always)]
+    unsafe fn read_account(&mut self) -> MaybeAccount {
+        let account: *mut Account = self.input as *mut Account;
+        // rent epoch (8 bytes)
+        self.input = self.input.add(core::mem::size_of::<u64>());
```

Comment:
This confused me a bit, since it's adding the rent epoch for a non-dupe account, and adding the dupe-marker + padding for a dupe account. And it's adding the rent-epoch out of order, which made me think it was handling the rent-epoch of the *previous* account.

I know it adds more code, but just for the sake of legibility, can you move this line to where it was before?

Reply:
It turns out that avoiding duplicating the line, it saves `1` CU per account. Added a comment to make it more clear what the offset it representing.